fix styles for hub pages and article page

marcochavezf opened the pull request:
cc @roryabraham @michelle-thompson 

### Details
This PR includes mainly style changes for the Help site to mimic the mockups design from the [design doc](https://docs.google.com/document/d/1dyJcO08hFD7XVhX92cMhay1qGp3DOK6z3RlelLA9b_4/edit#bookmark=id.ktp0ym5yhehl).

Some other minor changes included in this PR:
- Update outdated links from comments in the README.md file
- Rename files from the `_includes` folder (changed `-navigation-tree` suffix to `-lhn`)
- Add functionality to toggle the menu in small views (mobile web)

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/216742

### Tests
0. (you need ruby and bundler installed to do this, more info [here](https://github.com/Expensify/App/tree/main/docs#pre-requisites))
1. Run:

    ```bash
    cd docs
    bundle install
    bundle exec jekyll serve
    ```
2. Make sure the site's styles load and that all the colors look correct.

Additionally add more content like Stage titles, Groups of cards and card buttons to one of the hub pages (i.e. [request-money](https://github.com/Expensify/App/blob/5773b44613debc4dd1d9eddc1522b312f379b8a9/docs/hubs/request-money.html) and Stage title anchors and Article links to the respective LHN section (i.e. [request-money-lhn](https://github.com/Expensify/App/blob/5773b44613debc4dd1d9eddc1522b312f379b8a9/docs/_includes/request-money-lhn.html)).

### Screenshots

#### Hub page

Desktop (updated):
![localhost_4000_hubs_request-money_](https://user-images.githubusercontent.com/6829422/178348959-1d53e522-5c3a-4fe6-9ca9-6f1ffeb91053.png)

Mobile (updated):
<img src="https://user-images.githubusercontent.com/6829422/178124328-d4f6a80f-fb2e-440f-bc60-9522e2e9fa32.png" width="400" />
<img src="https://user-images.githubusercontent.com/6829422/178349010-775e9606-b887-4dfe-ba4f-96b63fc53b8d.png" width="400" />

#### Article page

Desktop:
![localhost_4000_articles_request-money_SmartScan](https://user-images.githubusercontent.com/6829422/178124343-4f8246a9-c0ed-42f2-8304-74f9c277e5f5.png)

Mobile:
<img src="https://user-images.githubusercontent.com/6829422/178124359-f9080658-64e5-490a-be7d-f31c9287e0c3.png" width="400" />

<img src="https://user-images.githubusercontent.com/6829422/178124363-de7d59bb-cd76-4c9b-8541-7d3b72a04baa.png" width="400" />

marcochavezf committed:
fix conflicts

marcochavezf committed:
fix grids with one element

marcochavezf committed:
fix logo styles

marcochavezf committed:
fix lhn styles

marcochavezf committed:
fix lhn styles for ul

marcochavezf committed:
fix h1 and content-area styles

marcochavezf committed:
fix heading styles

marcochavezf committed:
add scrollbar funcitonality to LHN

marcochavezf committed:
update lhn width for different sizes

marcochavezf committed:
add reponsiveness to the LHN

marcochavezf committed:
fix style for lhn header

marcochavezf committed:
fix styles for article content page

marcochavezf committed:
remove help site

marcochavezf committed:
fix color variables

marcochavezf committed:
fix back button for article lhn

marcochavezf committed:
fix spaces for main.scss

marcochavezf committed:
fix toc in lhn styles

marcochavezf committed:
update html files with latest changes

marcochavezf committed:
update titles

marcochavezf committed:
fix scroll in lhn

marcochavezf committed:
fix scrolling in mobile

marcochavezf committed:
clean scroll logic for lhn

marcochavezf committed:
fix position of back button

marcochavezf committed:
fix logo in hub content

marcochavezf committed:
fix icons in lhn

marcochavezf committed:
fix scroll for toc library

marcochavezf committed:
fix navigate back in mobile

marcochavezf committed:
reorganize lhn files

marcochavezf committed:
clean css code and fix caret icons glitch

marcochavezf committed:
clean css code for links

marcochavezf committed:
fix navigation back function

marcochavezf committed:
clean html content and update links in readme file

marcochavezf committed:
fix lhn height for mobile web

marcochavezf committed:
last minor fix to lhn

MariaHCD commented on docs/hubs/request-money.html:
```diff
@@ -27,7 +29,7 @@ <h2 id="request-money">
             <div class="title">SmartScan</div>
```
How about changing the cursor to a pointer so that it indicates the card is clickable

![Screen Shot 2022-07-11 at 5 09 23 PM](https://user-images.githubusercontent.com/12268372/178271621-df99e5ff-5cfb-4893-9069-0087bcc13fa7.png)

marcochavezf replied:
Good catch Maria! Updating it

MariaHCD commented on docs/_includes/home-lhn.html:
```diff
@@ -0,0 +1,13 @@
+        <a href="/hubs/send-money">Send money</a>
+    </li>
+    <li class="icon-with-link">
+        <i class="fa-solid fa-angle-right icon"></i>
```
I was expecting that clicking on the right caret icon would allow me to navigate the docs as well. What do you think?

![Screen Shot 2022-07-11 at 4 55 42 PM](https://user-images.githubusercontent.com/12268372/178269945-c4fe1b0a-a132-4090-93ad-08744c16d35b.png)

marcochavezf replied:
Good point, making the caret icons also clickable and adding the `cursor: pointer` value to them

MariaHCD commented on docs/_layouts/default.html:
```diff
@@ -7,33 +7,40 @@
-                        <i class="fa-solid fa-angle-left caret-icon"></i>Back
-                    </div>
+                <div onclick="navigateBack()" class="icon-with-link">
+                    <i class="fa-solid fa-angle-left icon"></i>
```
Probably can be a pointer cursor here too?

![Screen Shot 2022-07-11 at 5 24 13 PM](https://user-images.githubusercontent.com/12268372/178274724-78f067b9-ea42-4398-b751-1da1df875d05.png)

marcochavezf replied:
👍🏽

MariaHCD reviewed (commented):
Looks great! Just a few comments.

michelle-thompson commented:
<img width="380" alt="image" src="https://user-images.githubusercontent.com/43361905/178300945-73763333-aefb-4163-abbd-09c314869047.png">
The icon is the wrong color - it should be the same as the icons in the LHN. Additionally, the border is too dark - let's change it to #ECECEC

marcochavezf committed:
change cursor to a pointer so that it indicates the card is clickable

marcochavezf committed:
make caret icon also clickable

marcochavezf committed:
fix styles for card button in hub pages

marcochavezf committed:
fix mobile header buttons

marcochavezf commented:
> <img alt="image" width="380" src="https://user-images.githubusercontent.com/43361905/178300945-73763333-aefb-4163-abbd-09c314869047.png">
> 
> The icon is the wrong color - it should be the same as the icons in the LHN. Additionally, the border is too dark - let's change it to #ECECEC

Good catch @michelle-thompson! I updated the styles for the card in the hub pages and also updated the screenshots in the PR description.

@MariaHCD I addressed the requested changes, thanks for the review! 🙇🏽 The PR is ready to be reviewed again.

michelle-thompson commented:
One more comment, I noticed the hyperlinks are medium weight. Links should be regular weight (unless the body text is bolded, in which case it should follow the body text)
<img width="518" alt="image" src="https://user-images.githubusercontent.com/43361905/178372558-74ff7cca-b33c-436c-ae03-b452f5ae264e.png">

marcochavezf committed:
remove bold to link tags

marcochavezf commented:
> One more comment, I noticed the hyperlinks are medium weight. Links should be regular weight (unless the body text is bolded, in which case it should follow the body text) <img alt="image" width="518" src="https://user-images.githubusercontent.com/43361905/178372558-74ff7cca-b33c-436c-ae03-b452f5ae264e.png">

Ah yes, it was a bolded hyperlink, I think I accidentally added the bold property in the article .md content. This how the link looks like without the bold property:

![localhost_4000_articles_request-money_SmartScan (1)](https://user-images.githubusercontent.com/6829422/178580501-b1ebdf3e-a671-4044-b5ef-529e7b21a3a7.png)

michelle-thompson commented:
Looks great, thank you!

ctkochan22 commented:
Reviewing the other PR: https://github.com/Expensify/App/pull/9869#event-6981715901

It looks like we are good right? It seems @michelle-thompson gave it the thumbs up.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2694130502) to production by @chiragsalian in version: 1.1.84-13 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
